src/aragorn_app.py

```diff
@@ -102,56 +134,42 @@ async def subservice_callback(response: PDResponse,  guid: str) -> int:
     # init the return html status code
     ret_val: int = 200
 
-    logger.info(f'{guid}: Receiving sub-service callback')
+    logger.info(f"{guid}: Receiving sub-service callback")
     # logger.debug(f'{guid}: The sub-service response: {response.json()}')
 
-    # init the connection
-    connection = None
-
     try:
-        # create a connection to the queue
-        connection = await aio_pika.connect_robust(f"amqp://{q_username}:{q_password}@{q_host}/")
-
-        # with the connection post to the queue
-        async with connection:
-            # get a channel to the queue
-            channel = await connection.channel()
-
-            # only store file if there's a queue waiting for it
-            # will throw error if queue doesn't exist
-            await channel.get_queue(guid, ensure=True)
```

Comment:
Is the right way to ensure a queue exists?  Shouldn't it be something like: 

                await channel.declare_queue(guid, durable=False, passive=True, auto_delete=False)

Reply:
After reading the documentation for both methods, I think both are valid.  I will add that line back in.

Reply:
We don't want to create a new queue here, we want to make sure that one already exists before we write a new file to disk. If the queue doesn't exist, that means the original worker that's waiting for this response is no longer waiting for it and no one is going to read and then delete this file, so it's basically a storage space leak. What I was trying to do here was to error out if the queue didn't exist already.

Reply:
From the documentation:

async get_queue(name: str, *, ensure: bool = True) → AbstractQueue[[source]](https://aio-pika.readthedocs.io/en/latest/_modules/aio_pika/channel.html#Channel.get_queue)

With ensure=True, it’s a shortcut for .declare_queue(..., passive=True); otherwise, it returns a queue instance without checking its existence.

Reply:
Their documentation is quite bad. If you read the very next sentence under that one: When the queue does not exist, if ``ensure=True``, will raise :class:`aio_pika.exceptions.ChannelClosed`.